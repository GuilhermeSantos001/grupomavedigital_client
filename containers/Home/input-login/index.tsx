import React from 'react';
import { useRouter } from 'next/router';

import { Form, Input, Row, Col, Image, message } from 'antd';
import { LockOutlined, GlobalOutlined } from '@ant-design/icons';

import ButtonStyled from '@/containers/_global/_styles/_button';
import ButtonAlternateStyled from '@/containers/_global/_styles/_buttonAlternate';

import UserService from '@/services/user';
import useUser from '@/atom/user'

declare interface Props {
  toggleIsRegister: () => void;
  toggleIsLogged: () => void;
}

export default function InputLogin({
  toggleIsRegister,
  toggleIsLogged,
}: Props) {
  const { loading, login } = UserService();

  const router = useRouter();
  const user = useUser();

  const onFinish = (values: { email: string; password: string; }) => {
    login(values.email, values.password)
      .then((data) => {
        toggleIsLogged();
        user.set(data);
        router.push('/dashboard');
      })
      .catch((error) => message.error(error.message))
  };

  return (
    <Form
      name="normal_login"
      className="login-form h-[100vh] flex items-center justify-center"
      onFinish={onFinish}
    >
      <Row gutter={24} className={'w-[100%] lg:w-[80%]'}>
        <Col span={24} className={'my-2'}>
          <div style={{ display: 'flex', alignItems: 'center', justifyContent: 'center', padding: 5 }}>
            <Image
              alt='MergeOps Logo'
              preview={false}
              src="/logo_light.svg"
            />
          </div>
        </Col>
        <Col span={24}>
          <Form.Item
            name="email"
            rules={[
              {
                required: true,
                message: 'Por favor, insira seu e-mail!',
              },
            ]}
          >
            <Input prefix={<GlobalOutlined className="site-form-item-icon" />} type="email" placeholder="Email" />
          </Form.Item>
        </Col>
        <Col span={24}>
          <Form.Item
            name="password"
            rules={[
              {
                required: true,
                message: 'Por favor, insira sua senha!',
              },
            ]}
          >
            <Input
              prefix={<LockOutlined className="site-form-item-icon" />}
              type="password"
              placeholder="Senha"
            />
          </Form.Item>
        </Col>
        <Col span={24}>
          <Form.Item>
            <div className='flex flex-col items-center justify-center'>
              <ButtonStyled
                type="primary"
                htmlType="submit"
                loading={loading}
              >
                Conectar
              </ButtonStyled>
              <ButtonAlternateStyled
                type="primary"
                loading={loading}
                className="mt-3"
                onClick={toggleIsRegister}
              >
                Registrar-se
              </ButtonAlternateStyled>
            </div>
          </Form.Item>
        </Col>
      </Row>
    </Form>
  );
};
