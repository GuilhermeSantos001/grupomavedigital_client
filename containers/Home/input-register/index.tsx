import React from 'react'
import PhoneInput from 'react-phone-input-2'

import { Form, Input, Row, Col, message } from 'antd'
import {
  LockOutlined,
  GlobalOutlined,
  BankOutlined,
  SecurityScanOutlined
} from '@ant-design/icons'

import * as Styles from '@/containers/Home/input-register/styles'

import ButtonStyled from '@/containers/_global/_styles/_button'
import ButtonAlternateStyled from '@/containers/_global/_styles/_buttonAlternate'

import Messages from '@/containers/Home/input-register/messages'
import UserService from '@/services/user'

import PatternUsername from '@/utils/validators/pattern-username'
import PatternEmail from '@/utils/validators/pattern-email'
import PatternDomain from '@/utils/validators/pattern-domain'

declare interface Props {
  toggleIsRegister: () => void;
  buttonLogin?: boolean;
}

export default function InputRegister({
  toggleIsRegister,
  buttonLogin = true,
}: Props) {
  const { loading, register } = UserService();

  const [form] = Form.useForm();

  const onFinish = (values: {
    user_username: string;
    user_email: string;
    user_phone: string;
    user_password: string;
    company_name: string;
    company_domain: string;
    company_subdomain: string | undefined;
  }) => {
    register(
      values.user_username,
      values.user_email,
      values.user_password,
    )
      .then((user) => {
        message.success(`Bem-vindo(a) ${user.username}!. Verifique seu e-mail para ativar sua conta.`);
        toggleIsRegister();
      })
      .catch((error) => message.error(error.message))
  };

  return (
    <Form
      form={form}
      name="normal_register"
      className="register-form flex flex-col items-center justify-center"
      layout='vertical'
      onFinish={onFinish}
    >
      <Row gutter={24} className="w-[100%] lg:w-[50%]">
        <Col span={24}>
          <div className='p-5' style={{ display: 'flex', alignItems: 'center', justifyContent: 'center' }}>
            <p className='text-white text-xl lg:text-[2rem] font-bold'>
              Crie sua conta
            </p>
          </div>
        </Col>
        <Col span={24}>
          <Styles.FormItemLight
            name="user_username"
            label="Nome de usuário"
            rules={[
              {
                required: true,
                pattern: PatternUsername,
                message: Messages.emptyOrInvalidField
              },
            ]}
          >
            <Input prefix={<GlobalOutlined className="site-form-item-icon" />} type="text" placeholder="Escreva seu nome..." />
          </Styles.FormItemLight>
        </Col>
        <Col span={24}>
          <Styles.FormItemLight
            name="user_email"
            label="Endereço de Email"
            rules={[
              {
                required: true,
                pattern: PatternEmail,
                message: Messages.emptyOrInvalidField
              },
            ]}
          >
            <Input prefix={<GlobalOutlined className="site-form-item-icon" />} type="text" placeholder="seunome@meudominio.com.br" />
          </Styles.FormItemLight>
        </Col>
        <Col span={24} lg={12}>
          <Styles.FormItemLight
            name="user_password"
            label="Senha"
            rules={[
              {
                required: true,
                message: 'Por favor, insira sua senha!',
              },
            ]}
          >
            <Input
              prefix={<LockOutlined className="site-form-item-icon" />}
              type="password"
              placeholder="********"
            />
          </Styles.FormItemLight>
        </Col>
        <Col span={24} lg={12}>
          <Styles.FormItemLight
            name="user_password_confirmation"
            label="Confirme sua senha"
            rules={[
              {
                required: true,
                message: 'Por favor, confirme sua senha!',
                validator: (rule, value, callback) => {
                  if (value && value !== form.getFieldValue('user_password')) {
                    callback('As senhas não coincidem!');
                  }
                  callback();
                }
              },
            ]}
          >
            <Input
              prefix={<LockOutlined className="site-form-item-icon" />}
              type="password"
              placeholder="********"
            />
          </Styles.FormItemLight>
        </Col>
        <Col span={24}>
          <Form.Item>
            <ButtonStyled type="primary" htmlType="submit" loading={loading}>
              Registrar
            </ButtonStyled>
            {
              buttonLogin &&
              <ButtonAlternateStyled
                type="primary"
                loading={loading}
                className="mt-3"
                onClick={toggleIsRegister}
              >
                Já tenho uma conta
              </ButtonAlternateStyled>
            }
          </Form.Item>
        </Col>
      </Row>
    </Form>
  )
};
