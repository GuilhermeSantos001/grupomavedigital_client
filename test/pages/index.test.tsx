import '@testing-library/jest-dom/extend-expect'

import React from 'react'
import { render } from '../testUtils'
import Home from '../../pages/index'

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Home />, {})
    expect(asFragment()).toMatchSnapshot()
  })

  it('Verifica se a div principal está com as propriedades corretas', () => {
    const { getByTestId } = render(<Home />, {})

    expect(getByTestId('div-container')).toHaveClass('p-2')
    expect(getByTestId('div-container')).toHaveStyle({
      fontFamily: 'Fira Code',
    })
  })

  it('Verifica se o texto inicial está com a mensagem certa, e as propriedades corretas', () => {
    const { getByTestId } = render(<Home />, {})

    expect(getByTestId('text-start')).toHaveClass('fw-bold')
    expect(getByTestId('text-start')).toHaveTextContent('Bem-Vindo(a)')
  })

  it('Verifica se o divisor de linha primario está presente', () => {
    const { getByTestId } = render(<Home />, {})

    expect(getByTestId('separator')).toHaveClass('text-muted')
  })

  it('Verifica se o card 1 está com o layout correto', () => {
    const { getByTestId } = render(<Home />, {})

    expect(getByTestId('container-card-1')).toHaveClass('col-12 p-2')
    expect(getByTestId('card-image-1')).toHaveClass('card shadow')
    expect(getByTestId('image-1')).toHaveClass(
      'cursor-pointer',
      'card-img-top',
      'border',
      'border-primary'
    )
    expect(getByTestId('card-body-1')).toHaveClass(
      'card-body',
      'bg-primary',
      'bg-gradient',
      'text-secondary',
      'overflow-auto'
    )
    expect(getByTestId('card-body-1')).toHaveStyle({ height: '10rem' })
    expect(getByTestId('card-body-1-title')).toHaveClass('card-title')
    expect(getByTestId('card-body-1-title')).toHaveTextContent(
      'Nosso principal objetivo com essa renovação digital é integrar o que temos de mais importante, você!'
    )
    expect(getByTestId('card-text-1')).toHaveClass(
      'card-text',
      'text-white',
      'fs-6'
    )
    expect(getByTestId('card-text-1')).toHaveTextContent(
      'Queremos que nossos clientes tenham suas informações de qualquer lugar.'
    )
  })

  it('Verifica se o card 2 está com o layout correto', () => {
    const { getByTestId } = render(<Home />, {})

    expect(getByTestId('container-card-2')).toHaveClass(
      'd-flex flex-column flex-md-row'
    )

    expect(getByTestId('card-2-col-1')).toHaveClass('col-12 col-md-6 p-2')
    expect(getByTestId('card-2-1')).toHaveClass('card shadow')
    expect(getByTestId('image-2-1')).toHaveClass(
      'cursor-pointer card-img-top border border-primary'
    )
    expect(getByTestId('card-2-body-1')).toHaveClass(
      'card-body bg-primary bg-gradient text-secondary overflow-auto'
    )
    expect(getByTestId('card-2-title-1')).toHaveClass('card-title')
    expect(getByTestId('card-2-title-1')).toHaveTextContent('Use como preferir')
    expect(getByTestId('card-2-separator-1')).toHaveClass('text-white')
    expect(getByTestId('card-2-text-1')).toHaveClass(
      'card-text text-white fs-6'
    )
    expect(getByTestId('card-2-text-1')).toHaveTextContent(
      'O ambiente digital interativo foi pensado para ser flexível, você pode usar pelo computador, tablet ou smartphone.'
    )

    expect(getByTestId('card-2-col-2')).toHaveClass('col-12 col-md-6 p-2')
    expect(getByTestId('card-2-2')).toHaveClass('card shadow')
    expect(getByTestId('image-2-2')).toHaveClass(
      'cursor-pointer card-img-top border border-primary'
    )
    expect(getByTestId('card-2-body-2')).toHaveClass(
      'card-body bg-primary bg-gradient text-secondary overflow-auto'
    )
    expect(getByTestId('card-2-title-2')).toHaveClass('card-title')
    expect(getByTestId('card-2-title-2')).toHaveTextContent(
      'Capacita-se conosco'
    )
    expect(getByTestId('card-2-separator-2')).toHaveClass('text-white')
    expect(getByTestId('card-2-text-2')).toHaveClass(
      'card-text text-white fs-6'
    )
    expect(getByTestId('card-2-text-2')).toHaveTextContent(
      'Seja você um entusiasta ou um profissional do segmento, temos cursos para você aumentar suas habilidades.'
    )
  })
})
