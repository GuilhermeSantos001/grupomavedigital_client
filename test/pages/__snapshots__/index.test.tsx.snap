// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="p-2"
    data-testid="div-container"
    style="font-family: Fira Code;"
  >
    <h1
      class="fw-bold"
      data-testid="text-start"
    >
      Bem-Vindo(a)
    </h1>
    <hr
      class="text-muted"
      data-testid="separator"
    />
    <div
      class="col-12 p-2"
      data-testid="container-card-1"
    >
      <div
        class="card shadow"
        data-testid="card-image-1"
      >
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTkyMCIgaGVpZ2h0PSIxMDgwIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIvPg=="
              style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
            />
          </div>
          <img
            alt="Tecnologia & Inovação"
            class="cursor-pointer card-img-top border border-primary"
            data-nimg="intrinsic"
            data-testid="image-1"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </div>
        <div
          class="card-body bg-primary bg-gradient text-secondary overflow-auto"
          data-testid="card-body-1"
          style="height: 10rem;"
        >
          <h5
            class="card-title"
            data-testid="card-body-1-title"
          >
            Nosso principal objetivo com essa renovação digital é integrar o que temos de mais importante, você!
          </h5>
          <hr
            class="text-white"
            data-testid="card-separator-1"
          />
          <p
            class="card-text text-white fs-6"
            data-testid="card-text-1"
          >
            Queremos que nossos clientes tenham suas informações de qualquer lugar.
          </p>
        </div>
      </div>
    </div>
    <div
      class="d-flex flex-column flex-md-row"
      data-testid="container-card-2"
    >
      <div
        class="col-12 col-md-6 p-2"
        data-testid="card-2-col-1"
      >
        <div
          class="card shadow"
          data-testid="card-2-1"
        >
          <div
            style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="box-sizing: border-box; display: block; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTkyMCIgaGVpZ2h0PSIxMDgwIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIvPg=="
                style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
              />
            </div>
            <img
              alt="Notebook"
              class="cursor-pointer card-img-top border border-primary"
              data-nimg="intrinsic"
              data-testid="image-2-1"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </div>
          <div
            class="card-body bg-primary bg-gradient text-secondary overflow-auto"
            data-testid="card-2-body-1"
            style="height: 10rem;"
          >
            <h5
              class="card-title"
              data-testid="card-2-title-1"
            >
              Use como preferir
            </h5>
            <hr
              class="text-white"
              data-testid="card-2-separator-1"
            />
            <p
              class="card-text text-white fs-6"
              data-testid="card-2-text-1"
            >
              O ambiente digital interativo foi pensado para ser flexível, você pode usar pelo computador, tablet ou smartphone.
            </p>
          </div>
        </div>
      </div>
      <div
        class="col-12 col-md-6 p-2"
        data-testid="card-2-col-2"
      >
        <div
          class="card shadow"
          data-testid="card-2-2"
        >
          <div
            style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="box-sizing: border-box; display: block; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTkyMCIgaGVpZ2h0PSIxMDgwIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIvPg=="
                style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
              />
            </div>
            <img
              alt="Aula Online"
              class="cursor-pointer card-img-top border border-primary"
              data-nimg="intrinsic"
              data-testid="image-2-2"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </div>
          <div
            class="card-body bg-primary bg-gradient text-secondary overflow-auto"
            data-testid="card-2-body-2"
            style="height: 10rem;"
          >
            <h5
              class="card-title"
              data-testid="card-2-title-2"
            >
              Capacita-se conosco
            </h5>
            <hr
              class="text-white"
              data-testid="card-2-separator-2"
            />
            <p
              class="card-text text-white fs-6"
              data-testid="card-2-text-2"
            >
              Seja você um entusiasta ou um profissional do segmento, temos cursos para você aumentar suas habilidades.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
